tests/testInstanceCatalog.py

```diff
@@ -367,20 +376,93 @@ def testCannotBeNull(self):
                             if k < 4:
                                 if not np.isnan(xx):
                                     msg = ('k: %d -- %s %s -- %s' %
-                                           (k, str(xx), str(testData[j][k]), cat.cannot_be_null))
-                                    self.assertAlmostEqual(xx, testData[j][k], 3, msg=msg)
+                                           (k, str(xx), str(testData[ct_good][k]), cat.cannot_be_null))
+                                    self.assertAlmostEqual(xx, testData[ct_good][k], 3, msg=msg)
+                                else:
+                                    np.testing.assert_equal(testData[ct_good][k], np.NaN)
+                            else:
+                                msg = ('%s (%s) is not %s (%s)' %
+                                       (xx, type(xx), testData[ct_good][k], type(testData[ct_good][k])))
+                                self.assertEqual(xx.strip(), testData[ct_good][k].strip(), msg=msg)
+                        ct_good += 1
+
+                self.assertEqual(ct_good, len(testData))  # make sure that we tested all of the testData rows
+                msg = '%d >= %d' % (ct_good, ct_total)
+                self.assertLess(ct_good, ct_total, msg=msg)  # make sure that some rows did not make it into the catalog
+
+            if os.path.exists(fileName):
+                os.unlink(fileName)
+
+        def testCannotBeNull_pre_screen(self):
```

Comment:
I have changed test so that it both verifies that the catalogs produced with _pre_screen=True and _pre_screen=False are identical and that the size of _current_chunk is smaller with _pre_screen=True.